src/Extensions/Hosting.Services.Web/Middlewares/UserIdentity/UserIdentityMiddleware.cs

```diff
@@ -57,17 +57,23 @@ internal async Task<string> CreateUserHashAsync(HttpContext context)
 			foreach (IUserIdentityProvider provider in m_userIdentityProviders)
 			{
 				bool success;
-				(success, identityBytesWritten) = await provider.TryWriteBytesAsync(context, uidMemoryOwner.Memory.Span.Slice(0, provider.MaxBytesInIdentity))
+				(success, identityBytesWritten) = await provider.TryWriteBytesAsync(context, uidMemoryOwner.Memory)
 					.ConfigureAwait(false);
 				if (success)
 				{
 					break;
 				}
+				else
+				{
+					uidMemoryOwner.Memory.Span.Fill(0);
+				}
 			}
 
-			return GetHashString(m_saltProvider.GetSalt(), uidMemoryOwner.Memory.Span, identityBytesWritten);
+			return GetHashString(m_salt, uidMemoryOwner.Memory.Span, identityBytesWritten);
 		}
 
+		private ReadOnlySpan<byte> m_salt => m_saltProvider.GetSalt();
```

Comment:
I think it would be better to expose `MaxBytesInSalt` property on the `ISaltProvider` to avoid double calls to `GetSalt()`.

Reply:
Have added MaxBytesInSalt